src/nix/derivation-instantiate.cc

```diff
@@ -0,0 +1,93 @@
+#include "command.hh"
+#include "common-args.hh"
+#include "store-api.hh"
+#include "derivations.hh"
+#include "local-fs-store.hh"
+#include "progress-bar.hh"
+
+#include <nlohmann/json.hpp>
+
+using namespace nix;
+
+static nlohmann::json storePathSetToJSON(const StorePathSet & paths)
+{
+    auto res = nlohmann::json::array();
```

Comment:
Maybe make this an object and put `paths` as a key in there. This makes it extensible in future.